src/lib.rs

```diff
@@ -235,6 +256,7 @@ struct TestInfo {
     name: String,
     kind: String,
     is_ignored: bool,
+    ignored_message: Option<String>,
```

Comment:
I would prefer if we made the state `is_ignored: false, ignored_message: Some(_)` impossible to represent here. The state should not happen. So I'd suggest:

```suggestion
    ignored: Ignored,
```

And add a new enum:

```rust
enum Ignored {
    No,
    Yes {  msg: Option<String> },
}
```